src/lib/Scenes/Map/EventEmitter.ts

```diff
@@ -1,19 +1,41 @@
-export const EventEmitter = {
-  _events: {},
-  dispatch(event, data) {
-    if (!this._events[event]) {
+type Callback = (data: any) => void
+
+type Event = "map:change" | "filters:change"
```

Comment:
Just a thought but could be interesting to use an enum instead
```typescript
enum Event {
  MapChanged = "map:change"
  FiltersChanged = "filters:change"
}
```

Reply:
Ah yes, could be! For now I didn't want to change the API too much, as I’m not sure of the plans. Let me know if you feel strongly about doing so now.

Reply:
For sure, my reasoning as well. Let's leave it as is for now